framework/src/org/checkerframework/framework/util/CheckerMain.java

```diff
@@ -123,7 +135,17 @@ public void addToRuntimeBootclasspath(List<String> runtimeBootClasspathOpts) {
 
     protected List<String> createCpOpts(final List<String> argsList) {
         final List<String> extractedOps = extractCpOpts(argsList);
-        extractedOps.add(0, this.checkersJar.getAbsolutePath());
+        extractedOps.add(0, this.checkerQualJar.getAbsolutePath());
+        return extractedOps;
+    }
+
+    // Assumes that createCpOpts has already been run.
+    protected List<String> createPpOpts(final List<String> argsList) {
+        final List<String> extractedOps = extractPpOpts(argsList);
+        if (extractedOps.isEmpty()) {
+            extractedOps.addAll(this.cpOpts);
```

Comment:
I don't understand why the classpath is added to the processor path if no explicit processor path is found.  (Or maybe that's not what's happening here?)